lib/evmone/eof.cpp

```diff
@@ -579,6 +582,10 @@ EOFValidationError validate_eof1(evmc_revision rev, bytes_view main_container) n
         bytes_view bytes;
         bool referenced_by_eofcreate = false;
     };
+
+    if (main_container.size() > MAX_INITCODE_SIZE)
```

Comment:
An interesting case come up, concerning this piece: Should this check be done here (top-level `validate_eof`) or in `validate_header`.

An expectation would be to have the logic of `validate_header` protected against oversized containers. It has been since creation txs moved to the public API of eof.hpp, so, in principle, it should also contain the check as one of the first.

OTOH, this would be wasteful, as we'd be repeating the check for all subcontainers, which are always smaller than top-level. Also `validate_header` bare is only ever called in `host.cpp` during the process of validating a creation tx data (to be precise - to discover the split between container and calldata) - but there it is protected by the transaction-level MAX_INITCODE_SIZE check.

Let me know if you have any thoughts on this

Reply:
Does it mean the `validate_header` doesn't work correctly on its own?

Reply:
> Does it mean the `validate_header` doesn't work correctly on its own?

Only guaranteed for containers which have its size under the limit. In other words - checking size is outside of `validate_header`'s responsibility.

Reply:
If this is now the `validate_header()` precondition then add an assert in the `validate_header()`.